client/src/model/component/Connectable.hx

```diff
@@ -20,7 +20,7 @@ class Connectable extends CircuitElement implements Observer {
         this.connection.addObserver( this ) ;
     }
 
-    public function isPort() { return false ; }
+    public function isPort() { return Std.is(this, Port) ; }
```

Comment:
I thought this was ok the way it was.  Note that the method is overridden in Port.